src/Service/CallbackService.php

```diff
@@ -0,0 +1,104 @@
+	private bool $bIsStatic;
+
+	/**
+	 * @throws Exception
+	 */
+	public function __construct(string $sCallBackDefinition)
+	{
+		$this->sCallBackDefinition = $sCallBackDefinition;
+		if (stripos($sCallBackDefinition, '$this->') !== false) {
+			$this->sCallBackMethodName = str_ireplace('$this->', '', $sCallBackDefinition);
+			$this->bIsStatic = false;
+			$this->sCallBackClassName = ''; // we don't need it for non-static methods
+		} else {
+			if (!is_callable($sCallBackDefinition)) {
+				$sMessageError = "The callback '$sCallBackDefinition' is not valid.";
+				IssueLog::Error($sMessageError, 'console');
+				throw new Exception($sMessageError);
+			}
+			$iPos = strrpos($sCallBackDefinition, '::');
+			if ($iPos !== false && $iPos > 0) {
+				$this->sCallBackClassName = substr($sCallBackDefinition, 0, $iPos);
+				$this->sCallBackMethodName = substr($sCallBackDefinition, $iPos + 2);
+				$this->bIsStatic = true;
+			} else {
+				$sMessageError = "The callback '$sCallBackDefinition' is not a valid static method.";
+				IssueLog::Error($sMessageError, 'console');
+				throw new Exception($sMessageError);
+			}
+		}
+	}
+
+	public function IsStatic(): bool
+	{
+		return $this->bIsStatic;
+	}
+	/**
+	 * @throws ReflectionException
+	 * @throws SecurityException
+	 */
+	public function CheckCallbackSignature(string $sTriggeringObjectType, array $aParamsType): void
+	{
+		$iParamCount = 0;
+		if ($this->bIsStatic) { // If the callback is static, we expect the first parameter to be the triggering object type
+			array_unshift($aParamsType, $sTriggeringObjectType);
+		} else {
+			$this->sCallBackClassName = $sTriggeringObjectType;
+		}
+		$oReflector = new ReflectionClass($this->sCallBackClassName);
+		$aCallbackParameters = $oReflector->getMethod($this->sCallBackMethodName)->getParameters();
+		$iRequiredNumberOfParams = count($aParamsType);
+		if (count($aCallbackParameters) !== $iRequiredNumberOfParams) {
+			$sErrorMessage = "The callback method '$this->sCallBackMethodName' of class '$this->sCallBackClassName' must have exactly $iRequiredNumberOfParams parameters.";
+			IssueLog::Error($sErrorMessage);
+			throw new SecurityException($sErrorMessage);
+		}
+		foreach ($aCallbackParameters as $oParam) {
+			if ($oParam->getType() === null) {
+				$sErrorMessage = "The callback method '$this->sCallBackMethodName' of class '$this->sCallBackClassName' must have type-hinted parameters, but parameter {$oParam->getName()} is not type-hinted.";
+				IssueLog::Error($sErrorMessage);
+				throw new SecurityException($sErrorMessage);
+			}
+			if ($oParam->getType()->getName() !== $aParamsType[$iParamCount]) {
+				$sErrorMessage = "The callback method '$this->sCallBackMethodName' of class '$this->sCallBackClassName' must have a first parameter of type '$aParamsType[$iParamCount]', but it has {$oParam->getType()->getName()} instead.";
+				IssueLog::Error($sErrorMessage);
+				throw new SecurityException($sErrorMessage);
+			}
+			$iParamCount++;
+		}
```

Comment:
What do you think of this? Looping over expected params instead of given params..
```suggestion
		foreach ($aParamsType as $iParamOrder => $sParamType) {
			$oParam = $aCallbackParameters[$iParamOrder];
			if ($oParam->getType() === null) {
				$sErrorMessage = "The callback method '$this->sCallBackMethodName' of class '$this->sCallBackClassName' must have type-hinted parameters, but parameter {$oParam->getName()} is not type-hinted.";
				IssueLog::Error($sErrorMessage);
				throw new SecurityException($sErrorMessage);
			}
			if ($oParam->getType()->getName() !== $sParamType) {
				$sErrorMessage = "The callback method '$this->sCallBackMethodName' of class '$this->sCallBackClassName' must have a parameter of type '$sParamType', but it has {$oParam->getType()->getName()} instead.";
				IssueLog::Error($sErrorMessage);
				throw new SecurityException($sErrorMessage);
			}
		}
```